### ISSUE TITLE
Store training details on each Flair model

### ISSUE THREAD

**Issue by alanakbik** [flairNLP/flair#2454]
There are many details that influence how well a model works, including training parameters (optimizer, scheduler, mini-batch size, learning rate, etc.), the transformers version and the flair version used to train a model. Currently, these parameters are not explicitly written out (though some appear in the log files generated during training). 

This issue tracks the progress of saving these details directly in each trained model. Each model would therefore have an in-built "model card" that lists all relevant parameters used to train it. Likely this will also include the optimizer and scheduler states. This has two advantages: 
- You can retrieve for each model all parameters used to train it 
- You can continue training each model for more epochs starting from the last optimizer/scheduler state if you want
